app-frontend/src/app/core/services/scene.service.js

```diff
@@ -20,7 +22,11 @@ export default (app) => {
         }
 
         query(params = {}) {
-            return this.Scene.query(params).$promise;
+            let validParams = Object.assign(
+                params,
+                {minCloudCover: _.max([0, params.minCloudCover])}
```

Comment:
`Math.max([0, undefined]) = NaN`, while `_.max([0, undefined) = 0`. I subbed in a ternary expression to drop lodash instead